system/libsam/source/uotghs_host.c

```diff
@@ -360,7 +360,7 @@ void UHD_Pipe_Free(uint32_t ul_pipe)
 uint32_t UHD_Pipe_Read(uint32_t ul_pipe, uint32_t ul_size, uint8_t* data)
 {
 	uint8_t *ptr_ep_data = 0;
-	uint8_t nb_byte_received = 0;
+	uint16_t nb_byte_received = 0; //otherwise roll over when reading pipe >256 Bytes
```

Comment:
I feel the comment on this line will be confusing when read without the context of the other information in this pull request. If the comment is to be there, I think it could be better worded.

Reply:
You are right. 
Changed comment